novem/table/utils/format.py

```diff
@@ -86,3 +86,122 @@ def merge_from_index(src: Union[pd.DataFrame, pd.Index], io: Optional[int] = Non
             merge_instructions.append(f"{start_row + aio}:{row + aio} {level} lbl{len(merge_instructions) + 1}")
 
     return "\n".join(merge_instructions)
+
+
+try:
+    import pandas as pd
+except ImportError:
+    pd = None  # type: ignore
+
+
+def merge_from_index_first_rows(
+    src: Union[pd.DataFrame, pd.Index], io: Optional[int] = None, level: Optional[int] = None
+) -> str:
+    """
+    Get comma-separated list of first rows from each merge group at specified level.
+
+    Args:
+        src (Union[pd.DataFrame, pd.Index]): The source DataFrame or Index object to analyze.
+        io (Optional[int]): Initial offset. If provided, overrides the calculated offset.
+        level (Optional[int]): The level of the MultiIndex to analyze. If None, uses the
+            innermost (most granular) level.
+
+    Returns:
+        str: Comma-separated list of row numbers representing first rows of merged sections
+    """
+    if not isinstance(src, (pd.DataFrame, pd.Index)):
```

Comment:
I'm guessing this will crash if pandas is not available

Reply:
Well, the guards are there, so this works similar to the other pandas integrations. I would not expect users to import this utility function unless they were to use it with pandas. The tests are passing so that is good.

Perhaps we should add some non-pandas test environments as well.